src/ensembl/src/content/app/custom-download/components/checkbox-grid/CheckboxGrid.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+
+import styles from './CheckBoxGrid.scss';
+
+type Props = {
+  gridData: any;
+  columns: number;
+  hideUnchecked?: boolean;
+  hideTitles?: boolean;
+  checkboxOnChange: (status: boolean, subSection: string, id: string) => void;
+};
+
+export const filterCheckedAttributes = (attributes: any) => {
+  const filteredAttributes: any = {};
+
+  Object.keys(attributes).forEach((section) => {
+    Object.keys(attributes[section]).forEach((attributeId) => {
+      if (attributes[section][attributeId].checkedStatus === true) {
+        if (!filteredAttributes[section]) {
+          filteredAttributes[section] = {};
+        }
+        filteredAttributes[section][attributeId] =
+          attributes[section][attributeId];
+      }
+    });
+  });
+
+  return filteredAttributes;
+};
+
+export const getAttributesCount = (attributes: any) => {
+  let totalAttributes = 0;
+
+  if (!attributes || Object.keys(attributes).length === 0) {
+    return 0;
+  }
+  Object.keys(attributes).forEach((section) => {
+    totalAttributes += Object.keys(attributes[section]).length;
+  });
+
+  return totalAttributes;
```

Comment:
- can attributes be typed more precisely?
- what does !attributes mean (is it `undefined` or `null`)?
- depending on what !attributes mean, we can either use default argument (`{}`) for when there are no attributes or do something like `attributes = attributes || {}`
- the body of the function looks like a good candidate for a `reduce`